core/arch/riscv/include/riscv.h

```diff
@@ -33,7 +33,7 @@
 #define XRET			sret
 #endif
 
-#define CSR_MODE_BITS	SHIFT_U64(CSR_MODE_OFFSET, 8)
+#define CSR_MODE_BITS		(CSR_MODE_OFFSET << 8)
```

Comment:
Yes, that's included when referring the CSRs below in ASM files.

Reply:
We don't have an assembly version. End up with complication errors while compiling ASM files due to casting to `uint*_t`.

Reply:
https://github.com/OP-TEE/optee_os/blob/1d7bc98cf9e3924ad606ad813914dc23df79272e/lib/libutils/ext/include/util.h#L123

Reply:
Instead this https://github.com/OP-TEE/optee_os/blob/1d7bc98cf9e3924ad606ad813914dc23df79272e/lib/libutils/ext/include/util.h#L128 is called when performing read/write_csr() form c-code, this would evaluate to something like `csrrs a5,(((uint64_t)(1)<<(8))|0x000),a5`

Reply:
Sounds like something is very broken.